src/mrinufft/operators/base.py

```diff
@@ -279,7 +308,10 @@ def compute_smaps(self, method=None):
             Note that this callable function should also hold the k-space data
             (use funtools.partial)
         """
-        if isinstance(method, np.ndarray):
+        if isinstance(method, np.ndarray) or (
+            (self.backend == "torchkbnufft-cpu" or self.backend == "torchkbnufft-gpu")
+            and isinstance(method, torch.Tensor)
+        ):
```

Comment:
would rather do the check in the operator's init (and avoid the backend specific checks here)
```python 
# in __init__ of TorchKBnufft: 
if isinstance(smaps, torch.Tensor):
   self.smaps = smaps 
else: 
   self.compute_smaps(...) 
   self.smaps = torch.from_numpy(self.smaps) 
```